### ISSUE TITLE
Show active Launch configuration in RunToolbarAction

### ISSUE THREAD

**Issue by azoitl** [eclipse-platform/eclipse.platform#799]
One of the things that bothers my in my usage of Eclipse IDE is that I never know what will happen when I press the Run Button in the Toolbar. Quite often I start the wrong thing.  I noticed in other IDEs that there often information text is presented at the run button to give users feedback what will happen when the run button is pressed. I think this would also be great for Eclipse's RunToolbarAction.

For that I did first experiments and with little changes (i.e., add `mode="FORCE_TEXT" `to the action definition of RunToolbarAction) I could show some text in the toolbar. Where I'm now struggling is that when I try to update the text of the Action the action is moved as child to the drop-down. 

Therefore I would have two questions:

1. Is this a feature that should be present in the Eclipse Platform?
2. What is the correct way to change the Action text of a toolbar item with drop down.

If the answer to 1 is yes I would be happy to work on that issue. However I may would need some guidance on 2.

**Comment by akurtakov** [comment]
I use LaunchBar for this purpose (https://wiki.eclipse.org/CDT/LaunchBar). It comes from CDT but is general solution which satisfies my needs for that pretty good.

**Comment by azoitl** [comment]
Yes the CDT LaunchBar goes in the right direction. But for me it is a bit to intrusive and breaks with Eclipse UI. Therefore I was searching for something more lightweight.

**Comment by mickaelistria** [comment]
I'm afraid adding the launch name in the toolbar will be perceived as too intrusive by some users.
Would it be possible to start with adding the launch config name in the Hover text of the button?

**Comment by merks** [comment]
When it's working correctly, the hover does already say what it will run.  I too think that's sufficient...

**Comment by azoitl** [comment]
Yes the toolip works as expected. However I don't see a tooltip as a solution. Tooltips are requiring an active actions by a user. Which disturbs the current thinking process. Also I have to wait till it shows up. This is all very slow. 

However I understand that my proposed change would be to intrusive for some users.

The cool thing is that the infrastructure is very flexible and I would be able to do it just for our RCP. However there is IMHO a bug in JFace so that I can really do it. See  #[this discussion](https://github.com/eclipse-platform/eclipse.platform.ui/discussions/1273) for details.

**Comment by Bananeweizen** [comment]
You might be able to fix your original problem by simply not using the launch toolbar, but rather the launch configurations _view_ (not the dialog, the view). If find that both more predictable as well as scalable with larger amounts of launch configs.

**Comment by azoitl** [comment]
> You might be able to fix your original problem by simply not using the launch toolbar, but rather the launch configurations _view_ (not the dialog, the view). If find that both more predictable as well as scalable with larger amounts of launch configs.

@Bananeweizen this view is amazing and I will definitely will use it for my day to day Eclipse development. I could also imagine to use it for our RCP. Will need some tests and user feedback for that. Thx a lot. Even after 15 years of Eclipse usage I learn something new.

**Comment by merks** [comment]
I've never noticed that view either!  It would be cool if one could reorder things with drag and drop in that view...

**Comment by vogella** [comment]
@mduft contributed it a while ago, see https://eclipse.dev/eclipse/news/4.22/platform.php#launch-config-view

**Comment by SarikaSinha** [comment]
@azoitl 
Do you still feel the need of this issue or you are satisfied with Launch Configuration view?

**Comment by azoitl** [comment]
@SarikaSinha thanks for asking. Launch Configuration view is definitely a good thing. For my personal work and the RCP 4diac IDE from the Eclipse 4diac project I'm working on I definitely want to have more. Currently the main blocker for me is that somehow I see Toolbar Actions with Text broken: #1273. As pointed out I'm happy to work on it, but I would need a pointer to start.

**Comment by azoitl** [comment]
After discussing with many people about Eclipse IDE usability issues at OCX this week I can say that the current solution is a **usability nightmare.** Anything proposed here is intuitive and easy to use by new users neither by experienced users. Everyone I talked with told me they are not trusting the launch buttons. 

So the process most of our users are doing  is the following:
  1. try to move the mouse over the small block triangle
  2. click there
  3. locate the launch they need
  4. start it

If I take VSCode or IntelliJ IDEA as example there I select once the active launch config. It is clearly shown right beside the launch and debug button. If I press one of these or use the shortcut I always know what is happening. 
![image](https://github.com/user-attachments/assets/008b1d3b-431f-4935-801b-d1298132aacf)

During a normal day using Eclipse IDE I rarely want to switch the active launch. So to quote @mickaelistria as  _"adding the launch name in the toolbar will be perceived as too intrusive by some users" we all have to suffer daily!

cc @HeikoKlare @HannesWell @opcoach @BeckerWdf

**Comment by laeubi** [comment]
There is also a Plugin that shows the launch configs in a drow-down @mikerumpf is using that in their company but I can't remember the name.

Beside that if you hover the shortcut it shows the launch name already, so I think having a preferences to show it as a label (or not for those who like the old behavior) seems quite actionable.

Next you can define "favorites" and then access runs with a shortcut, I think this can be enhanced as well to mark one of the favorites the default that is always started regardless of what was current. All these seem to me non-breaking additions that could maybe improve the situation already.

**Comment by iloveeclipse** [comment]
There is also "launchbox" from CDT

**Comment by laeubi** [comment]
> launchbox

Do you probabbly mean https://wiki.eclipse.org/CDT/LaunchBar ?

**Comment by iloveeclipse** [comment]
Exact

**Comment by laeubi** [comment]
You can already hide unwanted items if you choose Window>Perspective>Customize Perspektive so as long as it is integrated there I think people have always the choice.

**Comment by azoitl** [comment]
I found the CDT LaunchBar always a bit to intrusive. It uses quite some space. However it could be a start from which we could refine. I'm also very open to other suggestions. 

As several people brought up Tooltips: I did several usability and user experience studies on IDEs. In all these studies with different people with different knowledge level a common feedback is always that Tooltips are very good for exploring an interface and learning more about it. But always I get also the feedback that Tooltips are not suitable for the daily work. Moving the mouse on an element and waiting for the Tooltip to appear takes to long. For information users constantly need other means have to be used.

**Comment by vogella** [comment]
I think this is implemented these days. Can you check if that is what you ask for?

<img width="407" height="112" alt="Image" src="https://github.com/user-attachments/assets/03658fbe-235c-435d-98a5-e2fe5c44bf38" />

**Comment by azoitl** [comment]
Unfortunately not. Because here I still have to click on the small arrow to see. I was looking more for something like CDT or, where I got my original inspiration from, IntellliJ IDEA has it:

<img width="719" height="187" alt="Image" src="https://github.com/user-attachments/assets/5dd1f923-ef36-43dc-aa0f-46e11d36dcde" />

I think I saw something similar also on VSCode but couldn't find an image quickly.
